lib/topics_filter.rb

```diff
@@ -101,6 +113,22 @@ def filter_status(status:, category_id: nil)
 
   def extract_and_validate_value_for(filter, values)
     case filter
+    when "activity-before", "activity-after", "created-before", "created-after",
+         "latest-post-before", "latest-post-after"
+      value = values.last
+
+      if value =~ /^(?<year>[12][0-9]{3})-(?<month>0?[1-9]|1[0-2])-(?<day>0?[1-9]|[12]\d|3[01])$/
+        year = $~[:year].to_i
+        month = $~[:month].to_i
+        day = $~[:day].to_i
+
+        return if day == 0 || month == 0 || day > 31 || month > 12
```

Comment:
I think we don't need that guard

<img width="459" alt="Screenshot 2023-04-27 at 11 36 52 am" src="https://user-images.githubusercontent.com/72780/234737780-b77a7aff-9b65-4de4-9bdb-a9ce1201caa4.png">

Reply:
Good point :+1: I've removed the guard.